scripts/netwarn.sh

```diff
@@ -1,29 +1,32 @@
 #!/bin/sh
+
+# Some OS's like Ubuntu with LightDM *IGNORE* the above shebang line when this
+# script is invcked from /usr/share/mate/autostart/netwarn-iiab-network.desktop
+#
+# WHAT HAPPENS: sh (dash) NOT BASH will always be run!  As confirmed by:
+#
+# ps -p $$    # Whereas 'echo $SHELL' DOES NOT show the actual running shell!
+#
+# RECAP: We hard-code the above '#!/bin/sh' for uniformity across all distros.
+
 if [ -f /etc/iiab/install-flags/iiab-network-complete ]; then
     exit
 fi
 
-zenity --question --width=350 --text="IIAB needs to configure networking:\n\n► Internet must be live before you begin.\n►You might be prompted for your password.\n\nContinue?"
-case $? in
-    -1|5)
-        exit 1
-        ;;
-
-    1)
-        exit 0
-        ;;
-
-    0)
-        x-terminal-emulator -e /usr/local/bin/iiab-network
-        ;;
-esac
+zenity --question --width=360 --text="IIAB needs to configure networking:\n\n► Internet must be live before you begin.\n►You might be prompted for your password.\n\nContinue?  (This can take 2-3 minutes)"
+rc=$?
+if [ "$rc" != "0" ]; then
+    exit $rc
```

Comment:
The ! here is what messes with Ubuntu-Mate-lightDM... Unless you have tested this back off and create your own version.

Reply:
`!=` is POSIX standard sh (dash) and definitely runs no problem on on MATE, RasPiOS both.

Clicking 'No' or 'Yes' both work correctly (to progress to the next stage anyway!)